File: new_kb/parser/parser_samples/sample.php
<?php

namespace Groundwork\Samples;

require_once 'helpers.php';

class Animal
{
    protected string $name;
    private int $age;

    public function __construct(string $name, int $age)
    {
        $this->name = $name;
        $this->age = $age;
    }

    // base implementation
    public function speak(): string
    {
        return "{$this->name} makes a sound.";
    }
}

class Dog extends Animal
{
    public function speak(): string
    {
        return "{$this->name} barks.";
    }
}

File: kb/parser/parser_samples/sample.rb
require 'json'

class Animal
  attr_accessor :name

  def initialize(name, age)
    @name = name
    @age = age
  end

  # base implementation
  def speak
    "#{@name} makes a sound."
  end
end

class Dog < Animal
  def speak
    "#{@name} barks."
  end
end

dog = Dog.new("Rex", 3)
puts dog.speak
